EmpireLegends/GameEngine.cpp

```diff
@@ -0,0 +1,132 @@
+#include "GameEngine.h"
+#include <iostream>
+#include <fstream>
+
+
+GameEngine::GameEngine()
+{
+	gameStart = nullptr;
+	startPhase = nullptr;
+	gameMap = nullptr;
+	mainGame = nullptr;
+	deck = nullptr;
+}
+
+GameEngine::~GameEngine()
```

Comment:
shouldn't the destructor also deal with `mainGame` and `deck`? as in to either delete them or re-assign them to `nullptr`?